### PR TITLE
[Feature][Docs] AWS Application Load Balancer (ALB) support

### LINKED ISSUE #657: [Feature][Docs] AWS Application Load Balancer (ALB) support
### Search before asking

- [X] I had searched in the [issues](https://github.com/ray-project/kuberay/issues) and found no similar feature requirement.

### Description

Some users are interested in ingress support of KubeRay. KubeRay has built-in ingress support. However, built-in ingress only supports simple Nginx setups (See [ingress.md#example-nginx-ingress-on-kind](https://github.com/ray-project/kuberay/blob/master/docs/guidance/ingress.md#example-nginx-ingress-on-kind) and https://github.com/ray-project/kuberay/pull/646 for more details). We do not recommend users use built-in ingress support in a production environment with complex routing requirements. Hence, we need to have more documentation about how to set up ingress without built-in ingress support.

AWS Application Load Balancer (ALB) is one of the most popular ingress controllers. It is good to have a document to describe how to set up ALB as an ingress controller and a load balancer for KubeRay.

### Use case

_No response_

### Related issues

#441 
#637

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by kevin85421**
## Why are these changes needed?
Some users are interested in ingress support of KubeRay. KubeRay has built-in ingress support. However, built-in ingress only supports simple Nginx setups (See [ingress.md#example-nginx-ingress-on-kind](https://github.com/ray-project/kuberay/blob/master/docs/guidance/ingress.md#example-nginx-ingress-on-kind) and https://github.com/ray-project/kuberay/pull/646 for more details). We do not recommend users use built-in ingress support in a production environment with complex routing requirements. Hence, we need to have more documentation about how to set up ingress without built-in ingress support.

AWS Application Load Balancer (ALB) is one of the most popular ingress controllers. It is good to have a document to describe how to set up ALB as an ingress controller and a load balancer for KubeRay.

**Why do not we recommend users use built-in ingress support in a production environment?**
* We cannot fulfill most use cases without exposing the whole ingress.
* Different ingress controllers have different features. For example, Nginx supports `rewrite-target`, and thus the built-in ingress support can support it. See #646 for more details. However, ALB does not support `rewrite-target` feature (https://github.com/kubernetes-sigs/aws-load-balancer-controller/issues/835), and thus it is hard to support it with the built-in ingress support.

## Related issue number
Closes #657 
#441 

## Checks

- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [x] Manual tests
   - [ ] This PR is not tested :(

## Screenshot
Access Ray Dashboard with the DNS name of the ALB.
Screen Shot 2022-10-26 at 1 34 27 PM

### CODE REVIEW

File: docs/guidance/ingress.md [hunk 1]
```diff
@@ -1,10 +1,16 @@
 ## Ingress Usage
 
+* KubeRay built-in ingress support: KubeRay will help users create a Kubernetes ingress when `spec.headGroupSpec.enableIngress: true`. Currently, the built-in support only supports simple Nginx setups. Note that users still need to install ingress controller by themselves. **We do not recommend users use built-in ingress support in a production environment with complex routing requirements.**
+  * [Example: Nginx Ingress on KinD (built-in ingress support)](#example-nginx-ingress-on-kind-built-in-ingress-support)
+
+* Set up ingress for KubeRay: We highly recommend users create Kubernetes ingress on their own.
```
Comment (DmitriGekhtman) [inline docs/guidance/ingress.md:6]:
```suggestion
* Manually setting up an ingress for KubeRay: **For production use-cases, we recommend taking this route.**
```

File: docs/guidance/ingress.md [hunk 2]
```diff
@@ -1,10 +1,16 @@
 ## Ingress Usage
 
+* KubeRay built-in ingress support: KubeRay will help users create a Kubernetes ingress when `spec.headGroupSpec.enableIngress: true`. Currently, the built-in support only supports simple Nginx setups. Note that users still need to install ingress controller by themselves. **We do not recommend users use built-in ingress support in a production environment with complex routing requirements.**
```
Comment (DmitriGekhtman) [inline docs/guidance/ingress.md:3]: nit: "users use" is a little redundant
Also, I generally prefer not to use the word "user" in documentation when possible. I think it's usually to either use "you" or to not directly refer to the reader/user at all. 

```suggestion
* KubeRay built-in ingress support: KubeRay will help users create a Kubernetes ingress when `spec.headGroupSpec.enableIngress: true`. Currently, the built-in support only supports simple Nginx setups. Note that users still need to install ingress controller by themselves. **We do not recommend using built-in ingress support in a production environment with complex routing requirements.**
```

**Review by DmitriGekhtman — APPROVED** [review]

File: docs/guidance/ingress.md [hunk 3]
```diff
@@ -62,4 +68,76 @@ kubectl describe ingress raycluster-ingress-head-ingress
...
+#
+# (2) Set the name of head pod service to `spec...backend.service.name`
+eksctl get cluster ${YOUR_EKS_CLUSTER} # Check subnets on the EKS cluster
+
+
+# Step4: Create an ALB ingress. When an ingress with proper annotations creates,
+#        AWS Load Balancer controller will reconcile a ALB (not in AWS EKS cluster).
+kubectl apply -f ray-operator/config/samples/alb-ingress.yaml
+
+# Step5: Check ingress created by Step4.
```
[the remark this answers is not in the document]
Comment (kevin85421) [reply docs/guidance/ingress.md]: Updated.

File: docs/guidance/ingress.md [hunk 4]
```diff
@@ -1,12 +1,18 @@
 ## Ingress Usage
 
+* KubeRay built-in ingress support: KubeRay will help users create a Kubernetes ingress when `spec.headGroupSpec.enableIngress: true`. Currently, the built-in support only supports simple Nginx setups. Note that users still need to install ingress controller by themselves. **We do not recommend using built-in ingress support in a production environment with complex routing requirements.**
```
Comment (DmitriGekhtman) [inline docs/guidance/ingress.md:3]: nit: Nginx -> NGINX

Comment (kevin85421) [reply docs/guidance/ingress.md]: Updated.

**Pull request merged**